.github/workflows/my-ci-pipeline.yml

```diff
@@ -0,0 +1,44 @@
+name: Continuous Integration
+on:
+  push:
+    paths-ignore:
+      - "README.md"
+  pull_request:
+    # Only on pull requests to main
+    branches:
+      - main
+    paths-ignore:
+      - "README.md"
+  schedule:
+    # Run every Sunday at 00:00 UTC
+    - cron: "0 0 * * 0"
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: echo Hello Alexandru
+
+      - name: build docker image
+        run: docker build --target test --tag test-image .
+
+      - name: run docker image unit tests
+        run: docker run --entrypoint poetry test-image run pytest todo_app/tests
```

Comment:
I had a little play with this and I think if we swapped the entrypoint in the Dockerfile to shell syntax (``["poetry", "run", "pytest"]``) then we would be able to just use `docker run test-image todo_app/tests` but this works nicely as is